lib/screens/home_screen.dart

```diff
@@ -58,4 +59,52 @@ class _HomeScreenState extends State<HomeScreen> {
       ),
     );
   }
+
+  void _openQrCodeDialog(BuildContext context) {
+    if (kIsWeb) return;
+    final peerService =
+        Provider.of<ChangeCallbackNotifier<PeerService>>(context, listen: false)
+            .callbackProvider;
+    if (!peerService.isServerRunning) {
+      showDialog(
+        context: context,
+        builder: (context) => AlertDialog(
+          title: const Text('The server is not running!'),
+          content: SingleChildScrollView(
+            child: ListBody(
+              children: const [
+                Text('Clients won\'t be able to connect to this device.'),
+                Text('Would you like to start the server now?'),
+              ],
+            ),
+          ),
+          actions: [
+            TextButton(
+              onPressed: () {
+                Navigator.pop(context);
+                peerService.startServer();
+                showDialog(
+                  context: context,
+                  builder: (context) => QrCodeDialog(),
+                );
+              },
+              child: Text('Yes'),
+            ),
+            TextButton(
+              onPressed: () {
+                Navigator.pop(context);
+                showDialog(
+                  context: context,
+                  builder: (context) => QrCodeDialog(),
+                );
+              },
+              child: Text('No'),
```

Comment:
Please change the code to close the modal, if a user selects "No".

Rational: I was confused to see the QR Code display even after selecting "No". Also, I don't see the value in displaying the QR Code without an active server.